public/js/user.js

```diff
@@ -256,21 +258,29 @@ function saveMarker(e) {
     site.type = type;
     site.longitude = pos.lng();
     site.latitude = pos.lat();
+    site.uid = currentUser;
     sites[id].siteData = site;
     sites[id].fadeGroup = getFadeGroup(site);
-    sites[id].infowindow.setContent(contentInfowindow(site));
-    sites[id].editwindow.setContent(contentEditwindow(site));
     sites[id].editwindow.close();
     sites[id].editBubbleOpen = false;
     sites[id].marker.setDraggable(false);
     var json = JSON.stringify(site);
     $.ajax({
-      url: "ping.php/atlas",
+      url: "/marker/"+(id.toString().length == 36?id:""),
```

Comment:
Try to avoid [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants) (the "36" here). Presumably, this is validating the string is likely a UUID because of the length; however, it is fragile (what if we switch to shorter marker IDs in the future?) and it spreads the business logic (i.e., ID validation) throughout the code, making it harder to maintain. If you want to validate the ID, that's admirable, but I'd suggest at least creating a small `isValidMarkerId()` function and invoking it here.

Also, even though I'm suggesting removing it from this line... just FYI: in general, it's better practice to use triple equals in your code unless you are specifically comparing complex types and want allow for type conversions. When in doubt, use `===`. 🙂

Reply:
I found some regex that matches uuids, so I used it to create the isValidMarkerId() function. :)

Thank you for the advice about `===`. I look up the differences between the 2 often, but I keep getting confused.